### Paramteres used by the maneuver_mpc node ###
maneuver_mpc:
  ros__parameters:
    # Solver parameters
    N:        10  # Prediction horizon
    dt:       0.1 # Sampling time
    # Cost function weights
    # Error penalty
    w_xe:     1e9 
    w_ye:     1e9
    w_psie:   1e9
    w_ue:     1e9
    w_ve:     1e9
    w_re:     1e9
    # Actuation penalty
    w_Fx:     15.0 
    w_Fy:     25.0
    w_Mz:     25.0
    # Actuation rate penalty
    w_Fx_dot: 1.0
    w_Fy_dot: 5.0
    w_Mz_dot: 5.0
    # Hydrodynamics coefficients
    Mx:       2641.9  # Total surge mass [rigid body mass + added mass] 
    My:       3084.0  # Total sway mass [rigid body mass + added mass]
    Mpsi:     915.5   # Total yaw inertia [rigid body inertia Izz + added inertia]
    LDx:      74.82   # Linear damping in surge
    LDy:      69.48   # Linear damping in sway
    LDpsi:    105.0   # Linear damping in yaw
    QDx:      748.22  # Quadratic damping in surge
    QDy:      992.53  # Quadratic damping in sway
    QDpsi:    523.27  # Quadratic damping in yaw
    # State constraints
    max_xy:   100.0
    max_psi:  3.14059265359
    max_u:    0.8
    max_v:    0.6
    max_r:    0.4
    # Actuation constraints
    max_Fx:   900.0
    max_Fy:   900.0
    max_Mz:   700.0



   

    
